main.cpp

```diff
@@ -125,419 +152,429 @@ struct CarWash
-    // Type
     char filterType;
-    // Envelope amount
     int envAmount;
-    // Wet/dry
     int wet;
 
     struct LFO
     {
-        //LFO frequency
         float frequency;
-        // bpm ratio of the LFO frequency
         int tempoMultiplier;
-        //LFO waveform
         int waveform;
-        //Makes waveform sharper/smoother
         float waveformMorph;
-        //tells whether lfo is synced to bpm
         bool bpmSync;
 
-        //change waveform
         void changeWaveform(int nextWave);
-        //change LFO bpm ratio
         void changeTempoMultiplier(int newValue);
-        //sync LFO to Oscillator
         void syncToOscillator(Oscillator syncTo, float warpAmount);
     };
 
-    // Switch state
     void changeType(char nextType);
-    //Switch inputs
     void switchInput(int newInput);
-    //Sweep up/down
-    void filterSweep(int startFreq, int endFreq);
+    void filterSweep(int startFreq, int endFreq, float sweepTimeInMillis);
 };
-/*
-Thing 7) Amplifier
-5 properties:
-    1) Drive level 
-    2) Tone
-    3) Waveshaper type
-    4) bass boost (int)
-    5) Output level 
-3 things it can do:
-    1) Attenuate signal
-    2) Saturation
-    3) Change waveshaper
- */
+
+void Filter::changeType(char nextType)
+{
+    filterType = nextType;
+}
+void Filter::switchInput(int newInput)
+{
+    newInput += 1;
+    if(newInput > 5) newInput = 0;
+}
+void Filter::filterSweep(int startFreq, int endFreq, float sweepTimeInMillis)
+{
+    for(float f = 0.0f; f < sweepTimeInMillis; f += 0.5f)
+    {
+        cutoffFreq = startFreq;
+        if(cutoffFreq < endFreq) cutoffFreq++;
+
+        else  cutoffFreq--;
+    } 
+}
+void Filter::LFO::changeWaveform(int nextWave)
+{
+    waveform = nextWave;
+}
+void Filter::LFO::changeTempoMultiplier(int newValue)
+{
+    tempoMultiplier = newValue;
+}
+void Filter::LFO::syncToOscillator(Oscillator syncTo, float warpAmount)
+{
+    frequency = syncTo.lfoAmount;
+    warpAmount = 30.3f;
+}
+
+
 struct Amplifier
 {
-    //Drive level 
     float driveLevel;
-    //Tone
     int toneLevel;
-    //Waveshaper type
     int waveshaperType;
-    // bass boost (int)
     float bassBoostLevel;
-    //Output level 
     float outputLevel;
 
-    //Attenuate signal
     void divideSignalBy(float denominator);
-    //Saturation
     void addDrive(float driveAmount, Filter inputFilter, Filter outputFilter);
-    //Change waveshaper
     void changeWaveshaperMode(int nextMode);
 };
-/*
-Thing 8) Delay
-5 properties:
-    1) Type 
-    2) Speed (ms)
-    3) Feedback
-    4) Tone
-    5) Wet/Dry
-3 things it can do:
-    1) Switch inputs
-    2) Switch feedback to be pre/post wet/dry
-    3) Process audio
- */
+void Amplifier::divideSignalBy(float denominator)
+{
+    outputLevel = outputLevel/denominator;
+}
+void Amplifier::addDrive(float driveAmount, Filter inputFilter, Filter outputFilter)
+{
+    outputFilter.cutoffFreq = inputFilter.cutoffFreq;
+    driveLevel += driveAmount;
+}
```

Comment:
When you learn about References, this will be perfectly fine code.  However, these are not references (and you can't use references yet because they haven't been covered).  these function parameters stop existing when the function ends, which means your changes to `outputFilter.cutoffFreq` aren't preserved.
- [ ] revise.  try making `outputFilter` a member variable of your Amplifier so the changes will be preserved.